File: kinotic-persistence/src/main/java/org/kinotic/persistence/api/model/idl/decorators/TextDecorator.java
package org.kinotic.persistence.api.model.idl.decorators;

import com.fasterxml.jackson.annotation.JsonIgnore;
import org.kinotic.idl.api.schema.StringC3Type;
import org.kinotic.idl.api.schema.decorators.C3Decorator;
import org.kinotic.idl.api.schema.decorators.DecoratorTarget;

import java.util.List;

/**
 * {@link TextDecorator} defines a {@link StringC3Type} as a text field
 * Created by Navíd Mitchell 🤪 on 5/14/23.
 */
public final class TextDecorator extends C3Decorator {

    @JsonIgnore
    public static final String type = "Text";

    public TextDecorator() {
        this.targets = List.of(DecoratorTarget.FIELD);
    }
}

File: kinotic-js/workspace/packages/os-api/src/api/model/idl/decorators/TextDecorator.ts
import { C3Decorator } from '@kinotic-ai/idl'

/**
 * Signals that a property is a text field, and will be full text indexed.
 */
export class TextDecorator extends C3Decorator {

    constructor() {
        super()
        this.type = 'Text'
    }
}
